desktop/cmp/persistenceManager/PersistenceManagerModel.ts

```diff
@@ -0,0 +1,345 @@
+                });
+            } catch (e) {
+                this.logError('Error applying persistWith', persistWith, e);
+                XH.safeDestroy(this._provider);
+                this._provider = null;
+            }
+        }
+
+        this.loadAsync();
+    }
+
+    // TODO - Carefully review if this method needs isStale checks, and how to properly implement them.
+    override async doLoadAsync(loadSpec: LoadSpec) {
+        const rawViews = await XH.jsonBlobService.listAsync({type: this.type, includeValue: true});
+
+        runInAction(() => (this.views = this.processRaw(rawViews)));
+
+        const {objects} = this;
+        // Auto-create an empty view if required
+        if (!objects.length) {
+            const newValue = await this.newObjectFn(),
+                newObject = await XH.jsonBlobService.createAsync({
+                    type: this.type,
+                    name: `My ${this.capitalNoun}`,
+                    value: newValue
+                });
+            runInAction(() => (this.views = this.processRaw([newObject])));
+        }
+
+        // Always call selectAsync to ensure pendingValue updated and onChangeAsync callback fired if needed
+        const id = this.selectedObject?.id ?? this.objects[0].id;
+        await this.selectAsync(id);
+    }
+
+    async selectAsync(id: StoreRecordId) {
+        this.selectedId = id;
+        if (!this.isDirty) return;
+
+        const {value} = this;
+
+        this.setPendingValue(value);
+        await this.onChangeAsync(value);
+
+        // If current value is empty, we know it is the auto-created default view -
+        // save it to capture the default state.
+        if (isEmpty(value)) {
+            await this.saveAsync(true);
+        }
+    }
+
+    async saveAsync(skipToast: boolean = false) {
+        const {token, id} = this.selectedObject;
+        if (this.isShared) {
+            if (!(await this.confirmShareObjSaveAsync())) return;
+        }
+        await XH.jsonBlobService.updateAsync(token, {
+            ...this.selectedObject,
+            value: this.pendingValue
+        });
+        await this.refreshAsync();
+        await this.selectAsync(id);
+
+        if (!skipToast) XH.successToast(`${capitalize(this.noun)} successfully saved.`);
+    }
+
+    async saveAsAsync() {
+        const {name, description} = this.selectedObject;
+
+        this.openSaveDialog({
+            name,
+            description,
+            value: this.pendingValue,
+            isAdd: false
+        });
+    }
+
+    async createNewAsync() {
+        const {name, description} = this.selectedObject,
+            newValue = await this.newObjectFn();
+
+        this.openSaveDialog({
+            name,
+            description,
+            value: newValue,
+            isAdd: true
+        });
+    }
+
+    @action
+    openSaveDialog(objStub: ObjStub) {
+        this.saveDialogModel = new SaveDialogModel(this, objStub);
```

Comment:
Interesting pattern here to create a new `SaveDialogModel` every time rather than one `readonly` model with an `isOpen` state